packages/flutter/lib/src/services/image_resource.dart

```diff
@@ -16,13 +16,14 @@ typedef void ImageListener(ui.Image image);
 /// image object might change over time, either because the image is animating
 /// or because the underlying image resource was mutated.
 class ImageResource {
```

Comment:
Let me look into this more, it does seem better than having the wonky read/write field (even with asserts).